dulwich/tests/test_index.py

```diff
@@ -369,7 +369,10 @@ def test_nonempty(self):
                 sorted(os.listdir(os.path.join(repo.path, 'c'))))
 
     def test_norewrite(self):
-        sync = getattr(os, 'sync', lambda: os.system('sync'))
+        if sys.platform != 'win32':
```

Comment:
Either seems reasonable. It's fine to have the test python3-only, I guess. Just so long as we don't run the test with a dummy os.sync, as that'll lead to spurious failures.